### ISSUE TITLE
Pytorch feature

### ISSUE THREAD

**Issue by Rachine** [mblondel/soft-dtw#4]
Hi ! 

Did you start working on the pytorch feature? Or should I start looking into it?
Thanks for sharing the code!

**Comment by mblondel** [comment]
@arthurmensch is also interested in this. I guess you two could collaborate!

**Comment by sculyi** [comment]
hi, i have the same problem, did you have any ideas?

**Comment by amirkargar** [comment]
Hi, did you guys have any progress on this?

**Comment by mblondel** [comment]
Closing since a PyTorch implementation is now available in https://github.com/arthurmensch/didyprog thanks to @arthurmensch.

**Comment by shaform** [comment]
It seems like a Pytorch implementation is not really available in that repo https://github.com/arthurmensch/didyprog/issues/5. So maybe this should be reopened?

**Comment by mblondel** [comment]
@shaform I agree. Does anyone want to tackle it? I won't be able to do it myself, since I don't have the use case myself. Here are the things to do:

- Create a pytorch function class (https://pytorch.org/docs/stable/notes/extending.html). This should be pretty straightforward (cf. chainer class for inspiration).
- Write a unit test that makes sure the gradient is correct.
- Port chainer example in examples/ to pytorch.

**Comment by lyprince** [comment]
@mblondel Here is my implementation (https://github.com/lyprince/sdtw_pytorch). 

I haven't written a proper demo yet, but if this is something people would like and would be relatively easy I could try to make time for it.

**Comment by mblondel** [comment]
Awesome! Thanks for sharing.

**Comment by Maghoumi** [comment]
@mblondel First, thanks again for your great paper!

I've been extensively using it for my research. I just publicly released a CUDA implementation of your algorithm for PyTorch that can run up to 100x faster than other available implementations. It's availble here: https://github.com/Maghoumi/pytorch-softdtw-cuda

Hope this helps others!
